plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts

```diff
@@ -1,29 +1,42 @@
 /**
  * External dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { STORE_KEY } from '~/marketing/data/constants';
 import { Plugin } from './types';
 
+const selector = 'getRecommendedPlugins';
 const category = 'marketing';
 
 type SelectResult = {
 	isLoading: boolean;
 	plugins: Plugin[];
+	removeRecommendedPlugin: ( slug: string ) => void;
 };
 
 export const useRecommendedPlugins = () => {
+	const { removeRecommendedPlugin } = useDispatch( STORE_KEY );
+
+	const callback = useCallback(
+		( slug ) => {
+			removeRecommendedPlugin( slug, category );
+		},
+		[ removeRecommendedPlugin ]
+	);
+
 	return useSelect< SelectResult >(
 		( select ) => {
 			const { getRecommendedPlugins, isResolving } = select( STORE_KEY );
 
 			return {
-				isLoading: isResolving( 'getRecommendedPlugins', [ category ] ),
+				isLoading: isResolving( selector, [ category ] ),
```

Comment:
Any purpose in assigning `getRecommendedPlugins` to a constant?  Does not seem to be a necessary change and adds time to find the which selector we're checking.

Side note and not 100% necessary to change in this PR if things are working okay, but `isResolving` is not quite as accurate as `hasFinishedResolution` for checking resolution status.

Reply:
> Side note and not 100% necessary to change in this PR if things are working okay, but isResolving is not quite as accurate as hasFinishedResolution for checking resolution status.

@joshuatf  Just for my curiosity, could you explain briefly the difference?

Reply:
Sure!  `isResolving` is a selector that checks the current state of a resolver, while `hasFinishedResolution` checks that a resolver has both begun and completed.

The subtle distinction is that `isResolving` for all selectors will initially be `false` until a resolver has kicked off.  This means that for a brief moment, it may appear that resolution is complete before it has even begun.  This can lead to FOUC and other unexpected behavior when we expect items to initially not be loaded.

There are more details on data store selectors in [this blog post](https://unfoldingneurons.com/2020/wordpress-data-store-properties-resolvers), but happy to answer any other questions around this that I can.

Reply:
> Any purpose in assigning `getRecommendedPlugins` to a constant? Does not seem to be a necessary change and adds time to find the which selector we're checking.

When I was developing it, I had `invalidateResolution( 'getRecommendedPlugins', [category] )`, and there are multiple `'getRecommendedPlugins'` strings, so I refactor it into a constant variable. Then, I removed `invalidateResolution` but didn't change this.

I have removed the constant variable and used inline string `'getRecommendedPlugins'` in [`029d4b7` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/029d4b775aae0bb0c6c061d59ac5e68d9b6c6412).

> Side note and not 100% necessary to change in this PR if things are working okay, but `isResolving` is not quite as accurate as `hasFinishedResolution` for checking resolution status.

Ahh yes, thanks for that reminder! I have fixed this in [`a3cde4b` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/a3cde4b91df9704ac5aef9e5a577164b6bb41613).